src/globus_cli/login_manager/manager.py

```diff
@@ -246,6 +293,9 @@ def get_gcs_client(
             collection_id=collection_id, endpoint_id=endpoint_id
         )
 
+        self.add_requirement(
+            gcs_id, scopes=[GCSEndpointScopeBuilder(gcs_id).manage_collections]
+        )
```

Comment:
This change doesn't seem obviously related to the other content of this PR, but I expect that it's needed to get the accurate scope passed to the client creds token callout.

Can you add a comment which explains? I don't think this impacts the non-client-creds case, but it would be good to have internal documentation about how things behave.

Reply:
Added a comment explaining